---
title: "Intro to Transcriptomics"
author: Nicole Black, Wade Boohar, Kayla Xu
date: 07/17/22
updated: 10/18/24
---

***Deliverables***
-Upload this R Notebook to your GitHub and submit the link to your Repo on Brightspace.
-Include any graphs or figures created in this assignment in the folder with your R notebook with descriptive file names.

Since this is an optional partner activity, it is okay if your answers are the same as your partnerâ€™s as long as everyone understands it and could explain it in their own words if asked. Each person must individually push their code to Github. *At the top of your R Notebook, write the name of you and your partner(s) as a comment.*

***Complete the following coding activity and answer any following questions as comments in your R Notebook***

In SummarizedExperiment Tutorial, you learned how to manipulate the SummarizedExperiment data structure and turn it into more readable dataframes, saving them as rna_counts, rna_clinical, and rna_genes. In this semi-guided assignment, you will use these dataframes to perform differential expression analysis based on tumor status.

*Pre-Assignment*
Use knitr function to set your working directory to your analysis_data folder in 490_cluster.
```{r setup}
 knitr::opts_knit$set(root.dir = normalizePath("/home1/neemuchw/490_cluster/analysis_data"))
```

If DESeq2 is not already installed, install it now
```{r}
if (!require("DESeq2", quietly = TRUE))
BiocManager::install("DESeq2")
```

Load in all necessary packages
```{r}
library(TCGAbiolinks)
library(SummarizedExperiment)
library(DESeq2)

rna_query <- GDCquery(project = "TCGA-SKCM", data.category = "Transcriptome Profiling", data.type = "Gene Expression Quantification", workflow.type = "STAR - Counts")
# GDCdownload(rna_query)
rna_se <- GDCprepare(rna_query)
```



*1*
Read in the rna_clinical, rna_genes, and rna_counts dataframes which you made in the "SummarizedExperiment Guided Tutorial" R Notebook

```{r}
# rna_clinical 
rna_clinical <- rna_se@colData 
rna_clinical <- as.data.frame(rna_clinical)

# rna_genes 
rna_genes <- rna_se@rowRanges@elementMetadata
rna_genes <- as.data.frame(rna_genes)

# rna_counts 
rna_counts <- rna_se@assays@data$unstranded
rna_counts <- as.data.frame(rna_counts)

```


*2*
In this assignment, you will run differential expression analysis comparing patient samples by whether the sample is from a tumor or normal tissue (this is the definition column in rna_clinical). You will need to choose a variable to control for covariance of: age and/or PAM50 subtype (paper_BRCA_Subtype_PAM50). 

Manipulate those columns so that they are ready for differential expression analysis (hint: what kind of variables are they? what data type are they by default? do you need to handle unknown values?) Filter out genes with a total expression across all patients less than 1000.
```{r}
# convert definition column to factor 
rna_clinical$definition <- as.factor(rna_clinical$definition)

# convert age to num
rna_clinical$age_at_diagnosis <- as.numeric(rna_clinical$age_at_diagnosis)

# convert PAM50 subtype to factor + handle missing values
rna_clinical$ajcc_pathologic_stage[is.na(rna_clinical$ajcc_pathologic_stage)] <- "Unknown"
rna_clinical$ajcc_pathologic_stage <- as.factor(rna_clinical$ajcc_pathologic_stage)

# filter out genes with total expression < 1000
gene_sums <- rowSums(rna_counts)
rna_counts_filtered <- rna_counts[gene_sums >= 1000, ]

# column names in rna_counts match row names in rna_clinical
rna_clinical <- rna_clinical[colnames(rna_counts_filtered), ]

```


*3*
Perform the differential expression analysis, All you need to do is fill in the appropriate # terms
```{r}
dds <- DESeqDataSetFromMatrix(countData = rna_counts_filtered,
                              colData = rna_clinical,
                              design= ~ age_at_diagnosis + definition)

dds_obj <- DESeq(dds)

resultsNames(dds_obj)

results <- results(dds_obj, format = "DataFrame", contrast = c(#definition, #Tumor, #Normal))

results <- data.frame(results)
```

Prepare results dataframe for EnhancedVolcano plotting. Add two columns, "-log10(padj)" and "gene_name". Fill in these columns appropriately.
```{r}
results <- results(dds, format = "DataFrame")
results <- data.frame(results)
results$log10_padj <- -log10(results$padj)
results$gene_name <- rownames(results)

```

*4*
Now we will use the EnhancedVolcano package to plot our results. The code is already completed and should run without adjustment if all code up to here is correct.
```{r}
EnhancedVolcano(results,
                lab = results$gene_name,
                x = 'log2FoldChange',
                y = '-log10(padj)',
                title = 'Sample Definition: Tumor vs Normal Tissue',
                pointSize = 1.0,
                labSize = 5.0)
```

*5*
# Explain what genes from each part of the Volcano Plot mean in terms of their significance and up/down regulation. 
top-right genes: These genes are strongly upregulated (high log2 fold change) and very  significant (low p-value).
bottom-right genes: These genes are upregulated, but their statistical significance is weak.
top-left genes:  These genes are strongly downregulated (low log2 fold change) and very statistically significant.
bottom-left genes: These genes are downregulated, but not statistically significant.
top-middle genes: These genes have low fold change (close to zero) but high statistical significance.
bottom-middle genes: These genes do not show much expression change, and their statistical significance is low.