---
title: "Introduction to MAF"
author: Mahija Mogalipuvvu, adapted from Nicole Black, Wade Boohar
date: 07/17/22
updated: 10/3/24
---

***Deliverables***
- upload this R Notebook to your GitHub and submit the link to your Repo on Brightspace
- include ALL graphs or figures created in this assignment in a folder with your R notebook with descriptive file names.

We encourage you to work with a partner. Therefore, it is okay if your answers are the same as your partner’s as long as everyone understands it and could explain it in their own words if asked. Each person must individually push their code to Github. *At the top of your R Notebook, write the name of you and your partner(s) as a comment.*

***Complete the following coding activity and answer any following questions as comments in your R Notebook***

In this assignment, you will need to use your skills learned in class to demonstrate your understanding of categorical variables and R data structures.

*Pre-Assignment*
Load all necessary packages, read in the clinical data.csv file you have in your analysis_data folder, and instantiate the MAF_object.
```{r}
library(BiocManager)
library(TCGAbiolinks)
library(maftools)
clin_query <- GDCquery(project = "TCGA-BRCA", data.category = "Clinical", data.type = "Clinical Supplement", data.format = 'BCR Biotab')

clinical.BCRtab.all <- GDCprepare(clin_query)
clinic <- clinical.BCRtab.all$clinical_patient_brca[-c(1,2),]
colnames(clinic)[ colnames(clinic) == "bcr_patient_barcode" ] <-"Tumor_Sample_Barcode"
maf_query <- GDCquery(project = "TCGA-BRCA", data.category = "Simple Nucleotide Variation", access = "open", data.type = "Masked Somatic Mutation", workflow.type = "Aliquot Ensemble Somatic Variant Merging and Masking")
GDCdownload(maf_query)
maf <- GDCprepare(maf_query)
maf_object <- read.maf(maf = maf,clinicalData = clinic, isTCGA = TRUE)
```


*1*
Choose a clinical variable (or any variable from clin_rad or clin_drug) to separate your populations into two different groups and rewrite the column or create a new column with that variable as a factor. ***Do not use age or vital_status as your clinical variable.*** Hint: if your variable is continuous, you will need to determine your own cutoffs for the different levels of the factor. If your variable is categorical and has more than two possible values, choose the two that are the most common.
```{r}
clinic$ER_status <- ifelse(clinic$er_status_by_ihc == "Positive", "ER+", "ER-")
clinic$ER_status <- as.factor(clinic$ER_status)
```


*2*
Create a co-oncoplot with the top 10-20 (you choose) most mutated genes for the two groups. Pick one that has a large discrepancy in % mutated or type of mutations between the groups and research it. 
Research it. What is the gene used for? Can you think of any reason for the discrepancy?
```{r}
maf_ERplus <- subsetMaf(maf_object, clinQuery = "ER_status == 'ER+'")
maf_ERminus <- subsetMaf(maf_object, clinQuery = "ER_status == 'ER-'")

# Create co-oncoplot for top 15 most mutated genes
coOncoplot(m1 = maf_ERplus, 
           m2 = maf_ERminus, 
           m1Name = "ER+", 
           m2Name = "ER-", 
           genes = 15)
```
One commonly observed gene with significant differences in mutation rates between tumor and tumor-free groups is TP53. TP53 mutations are frequently found in various cancers, with much higher prevalence in tumor samples compared to normal tissues. TP53 works in DNA damage repair halts the cell cycle to allow for DNA repair or triggers apoptosis if the damage is irreparable. A possibe reason for this discrepancy is that one commonly observed gene with significant differences in mutation rates between tumor and tumor-free groups is TP53. TP53 mutations are frequently found in various cancers, with much higher prevalence in tumor samples compared to normal tissues. TP53 works in DNA damage repair halts the cell cycle to allow for DNA repair or triggers apoptosis if the damage is irreparable.

*3*
Create a contingency table with your variable and chosen gene. Run a Fisher’s Exact Test between presence of mutations for that gene and your clinical variable. Create and save a mosaic plot. 
Interpret the output of the Fisher’s Exact Test in terms of the odds ratio and p-value.
```{r}
clinic$TP53_mutated <- ifelse(clinic$Tumor_Sample_Barcode %in% maf_object@gene.summary[maf_object@gene.summary$Hugo_Symbol == "TP53",]$Tumor_Sample_Barcode, "Mutated", "Wild-Type")
cont_table <- table(clinic$ER_status, clinic$TP53_mutated)
print(cont_table)

# Fisher's Exact Test
fisher_test <- fisher.test(cont_table)
print(fisher_test)

# mosaic plot
mosaicplot(cont_table, col = c("blue", "pink"), main = "TP53 Mutation and ER Status")
```
The results of Fisher’s Exact Test indicate that the p-value is 0.1432, which is above the typical significance threshold of 0.05. This suggests that there is no statistically significant association between the two groups (tumor-free and tumor-present) in the analyzed dataset.

The odds ratio is 1.469, meaning that tumor-free individuals are 1.47 times more likely to have the characteristic in question compared to individuals with tumors. However, since the p-value is not statistically significant, this observed difference could be due to chance rather than a meaningful biological association.

*4*
Subset your maf_object based on your chosen clinical variable and create a co-lollipop plot of your chosen gene divided between the two different clinical variable possibilities. Include descriptive names on your plot.
Do you notice any difference in terms of mutations (e.g. sites, types, number) between the two populations?
```{r}
lollipopPlot2(m1 = maf_ERplus, 
              m2 = maf_ERminus, 
              m1_name = "ER+", 
              m2_name = "ER-", 
              gene = "TP53")
```
The Tumor-Free group has a higher percentage of individuals with PIK3CA mutations (35.45%) compared to the Tumor group (27.16%), suggesting a potential difference in the role of PIK3CA mutations between the two populations. Additionally, the types and distribution of mutations appear similar, with missense mutations being the most common, but the Tumor-Free group has more mutations overall, which may indicate a different selective pressure or functional consequence in tumor development.

*5*
Create your Overall_Survival_Status column and create a mafSurvival KM plot based on mutations in your chosen gene.
Does there seem to be a difference? Hypothesize why or not based on the other analysis you did with the gene above.
```{r}
clinic$Overall_Survival_Status <- ifelse(clinic$vital_status == "Dead", 1, 0)
mafSurvival(maf = maf_object, 
            genes = "TP53", 
            time = "days_to_last_followup", 
            Status = "Overall_Survival_Status")
```
The survival plot shows no statistically significant difference between the mutant and wild-type (WT) groups, as indicated by the p-value of 0.586 and a hazard ratio of 0.883, which is close to 1. This suggests that PIK3CA mutations do not strongly impact overall survival in this dataset. One possible explanation for this is that PIK3CA mutations may not be the primary driver of tumor progression or aggressiveness but rather contribute to tumor initiation or maintenance. Additionally, since PIK3CA mutations were more frequent in tumor-free individuals, this could indicate that in some contexts, these mutations do not always lead to worse outcomes, or their effects may be counteracted by other genetic or environmental factors. This aligns with our earlier observation that PIK3CA mutations are more common in the tumor-free group, suggesting they may play a different role than typically expected in cancer progression.
